### ISSUE TITLE
Header Filters outside table

### ISSUE THREAD

**Issue by samshah777** [olifolkerd/tabulator#520]
Hey Oli,

Quick question: Is there a way to have the header filters sit outside the table? The idea is to bind them within a div with textboxes and style them

Regards,
Sam

**Comment by olifolkerd** [comment]
Hey, 

You cant put the header filters outside the table because there would be no way to track their position relative to the headers when the table columns were moved or resized.

You can however use any input select or other element outside of the table to trigger a filter. If you have a look at the [Filter Data Example](http://olifolkerd.github.io/tabulator/examples/3.2/#filter) on the website, it does exactly this and comes with some example source code

Let me know if that helps.

Cheers

Oli

**Comment by samshah777** [comment]
Thanks Oli..although in the Filter Data example, "sequential filtering" is not possible, hence does not apply to me. For example: What I want to do is first filter all males, then those males with colour red. 

I've been trying to figure out another way to do this..

**Comment by olifolkerd** [comment]
Hey,

You can adapt the example in the demo to do exactly that. 

You need one function that is triggered by a change on any of the inputs that then builds a set of filters and applies them to the table:

```js
function updateFilter(){
	var filters = [];

	//add gender filter to array
	filters.push({
		field:"gender", 
		type:"=", 
		value:$("#gender-input").val()
	});

	//add colour filter to array
	filters.push({
		field:"colour", 
		type:"=", 
		value:$("#colour-input").val()
	});

	$("#example-table").tabulator("setFilter", filters);
}

//trigger filter update on value change
$("#gender-input, #colour-input").on("change", function(){
    updateFilter();
});
```

Let me know if that helps

Cheers

Oli

**Comment by samshah777** [comment]
Ahh I understand what you mean now. That said, your code does not seem to be working. 
What am I doing wrong?

BODY:
`<input type="text" id="gender-input" style="position:absolute;top:100px;">`
`<input type="text" id="colour-input" style="position:absolute;top:170px;">`
`<div id="#example-table"></div>`

JAVASCRIPT:
//Your code above
//Clear Filter
//Build Tabulator
//Sample data
//load sample data

**Comment by samshah777** [comment]
Nope..
The table shows fine. Filter not getting triggered. Code seems syntactically correct.
Tried with the filter data example, and it works fine.

Regards,
Sam

**Comment by samshah777** [comment]
Maybe an issue with this line -
`$("#gender-input, #colour-input").on("change", function(){
    updateFilter();});`

Cant figure it out..

**Comment by olifolkerd** [comment]
could you post a link to a js fiddle with your code setup please

**Comment by samshah777** [comment]
Thanks Oli, I managed to implement it another way using the add and remove filter functionalities. 

What I found was that hard coded values into `$("#example-table").tabulator("setFilter", filters);` worked fine. The issue was that the push into the array filters did not render properly with the above line. 

Furthermore, as per your example in the documentation, setFilter as below
 `$("#example-table").tabulator("setFilter", [
    {field:"age", type:">", value:52}, //filter by age greater than 52
    {field:"height", type:"<", value:142}, //and by height less than 142
]);`
does not work, even with the example you posted. Don't know if its a bug or if I was doing something wrong..

Regards,
Sam
